Add a page for general contribution guidelines, with a code of conduct and AI disclaimer (#19)
godotengine/godot-contributing-docs: organization/general_rules_and_guidelines.rst

General rules and guidelines
============================

This page contains guidelines relevant for contributing to any area of Godot.

When contributing to any particular area, please refer to its respective guidelines additionally.

Code of conduct
---------------

When you contribute to Godot, we expect that you respect our `Code of Conduct <https://godotengine.org/code-of-conduct/>`__.

Contributing with AI and large language models
----------------------------------------------

The use of AI to contribute to Godot is discouraged.
We acknowledge that AI can be useful, but we are convinced that effort by humans results in better and more relevant
contributions.

If you use AI to assist with your contribution, you are required to:

- Understand and verify any of its solutions.
    - If you don't understand the solution, do not submit it.
- Proof read and edit anything it generates.
    - Better, use AI to proof read something *you* create.
- Disclose how and to what degree you used AI (in the issue description, pull request description, or otherwise).

>     - In particular, you must disclose if you used AI to write part of your contribution or issue / pull request description.

Author: I decided to rephrase your suggestion a bit. I wanted to avoid making it sound like we're welcoming people using AI to fill out our language gaps in weblate.

> Contributions made entirely by AI are prohibited.

Reviewer: "Prohibited" itself is an excessively strong word. We don't often ban users for their low-quality submissions, nor we have the authority to prevent them from submitting them in the first place (which is what "prohibit" implies in my opinion).

> Contributions made entirely by AI are prohibited.

Reviewer 2: I agree with Ivor here. Generally "prohibit" is much too strong of a word. But in this case we need to be extremely firm or else we might as well say nothing at all. 

Language like "we encourage you not to submit PRs made entirely by AI" is practically useless. 

Importantly, we aren't saying you can't submit a PR with some AI generated code. What is prohibited are PRs that are made entirely by AI without being validated

Perhaps an acceptable compromise would be:

"Contributions made entirely by AI without careful human testing and validation are prohibited."

General rules and guidelines
============================

This page contains guidelines relevant for contributing to any area of Godot.

When contributing to any particular area, please refer to its respective guidelines additionally.

Code of conduct
---------------

When you contribute to Godot, we expect that you respect our `Code of Conduct <https://godotengine.org/code-of-conduct/>`__.

Contributing with AI and large language models
----------------------------------------------

> The use of AI to contribute to Godot is discouraged.

Reviewer 2: ```suggestion
----------------------------------------------

.. note:: When this page refers to "AI" it means any LLM/generative
              AI model like ChatGPT, Clause, Grok, etc.  or any any other
              future models that fulfill the same role.

The use of AI to contribute to Godot is discouraged.
```

To address Micky's concerns about whether people will understand exactly what we mean by "AI"

We acknowledge that AI can be useful, but we are convinced that effort by humans results in better and more relevant

> contributions.

Reviewer: Since we want to use "prohibited" later I guess we can upgrade this statement a notch.
```suggestion
The use of AI to contribute to Godot is **highly discouraged**.
We acknowledge that AI can be useful, but we are convinced that human effort results in better and more relevant
contributions.
```

Reviewer: Then you may just say discouraged but still in bold

Author: What's the motivation behind making it bold?

Reviewer: Mmmm... ease of reading, I suppose. If someone wants the juice of this paragraph, it's highlighted in bold right there.

If you use AI to assist with your contribution, you are required to:

- Understand and verify any of its solutions.
    - If you don't understand the solution, do not submit it.
- Proof read and edit anything it generates.

>     - Better, use AI to proof read something *you* create.

Reviewer: I'd not use AI to even proofread if I'm going to be frank with you, but okay let's pretend it's better 😩
```suggestion
- Proofread and edit anything it generates.
    - Better, use AI to proofread something created on your own.
```
Although I'm not exactly sure what this is trying to imply

Reviewer: Then let's spell it out, it's probably better than attempting to summarise it like this.

> - Disclose how and to what degree you used AI (in the issue description, pull request description, or otherwise).

Reviewer: "or otherwise"?

Reviewer: What you may want to do is explain what "contribution" refers to in this case way beforehand (maybe below _"The use of AI to contribute to Godot is discouraged."_), which is actually quite vague up until these examples in parentheses. Assuming that's done:
```suggestion
- Disclose how and how much you used AI for your contribution.
```

Author: That's a great way to phrase it!

General rules and guidelines
============================

This page contains guidelines relevant for contributing to any area of Godot.

> When contributing to any particular area, please also refer to its respective guidelines.

Reviewer 3: ```suggestion
When contributing to any particular area, please also refer to their respective guidelines.
```

Code of conduct
---------------

When you contribute to Godot, we expect that you respect our `Code of Conduct <https://godotengine.org/code-of-conduct/>`__.

Contributing with AI and large language models
----------------------------------------------

.. note:: When this page refers to "AI" it means any LLM/generative
          AI model like ChatGPT, Claude, Grok, etc., or any other
          models that fulfill the same role.

The use of AI to contribute to Godot is discouraged.
We acknowledge that AI can be useful, but we are convinced that effort by humans results in better and more relevant
contributions.

If you use AI to assist with your contribution, you are required to:

- Understand and verify any of its solutions.
    - If you don't understand the solution, do not submit it.
- Proofread and edit anything it generates.

>     - Better, use AI to proofread something *you created*.

Reviewer: ```suggestion
    - Alternatively, it would be better to let AI proofread something you created *on your own*.
```
